src/infinite-hits/__tests__/infinite-hits.spec.ts

```diff
@@ -35,22 +35,103 @@ describe('InfiniteHits', () => {
+    const button = fixture.debugElement.nativeElement.querySelector(
+      '.ais-InfiniteHits-loadMore'
+    );
     button.click();
 
-    expect(button.disabled).toBeTruthy();
+    expect(button.disabled).toEqual(true);
     expect(showMore).not.toHaveBeenCalled();
     expect(fixture).toMatchSnapshot();
   });
+
+  it('should display `showPrevious()` button', () => {
+    const showPrevious = jest.fn();
+    const renderWithPrevious = createRenderer({
+      defaultState,
+      template: '<ais-infinite-hits [showPrevious]=true></ais-infinite-hits>',
+      TestedWidget: NgAisInfiniteHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = renderWithPrevious({ showPrevious });
+
+    const button = fixture.debugElement.nativeElement.querySelector(
+      '.ais-InfiniteHits-loadPrevious'
+    );
+
+    expect(button).not.toBeFalsy();
+  });
+
+  it('should display `showPrevious()` button with custom label', () => {
+    const showPrevious = jest.fn();
+    const renderWithPrevious = createRenderer({
+      defaultState,
+      template:
+        '<ais-infinite-hits [showPrevious]=true showPreviousLabel="Load previous"></ais-infinite-hits>',
+      TestedWidget: NgAisInfiniteHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = renderWithPrevious({ showPrevious });
+
+    const button = fixture.debugElement.nativeElement.querySelector(
+      '.ais-InfiniteHits-loadPrevious'
+    );
+
+    expect(button.innerHTML).toMatchInlineSnapshot(`" Load previous "`);
+  });
+
+  it('should call `showPrevious()` on button click', () => {
+    const showPrevious = jest.fn();
+    const renderWithPrevious = createRenderer({
+      defaultState,
+      template: '<ais-infinite-hits [showPrevious]=true></ais-infinite-hits>',
+      TestedWidget: NgAisInfiniteHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = renderWithPrevious({ showPrevious });
+
+    const button = fixture.debugElement.nativeElement.querySelector(
+      '.ais-InfiniteHits-loadPrevious'
+    );
+    button.click();
+
+    expect(button.disabled).toEqual(false);
+    expect(showPrevious).toHaveBeenCalled();
+  });
+
+  it('should disable `showPrevious` button', () => {
+    const showPrevious = jest.fn();
+    const renderWithPrevious = createRenderer({
+      defaultState,
+      template: '<ais-infinite-hits [showPrevious]=true></ais-infinite-hits>',
+      TestedWidget: NgAisInfiniteHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = renderWithPrevious({ showPrevious, isFirstPage: true });
+
+    const button = fixture.debugElement.nativeElement.querySelector(
+      '.ais-InfiniteHits-loadPrevious'
+    );
+    button.click();
+
+    expect(button.className).toMatchInlineSnapshot(
+      `"ais-InfiniteHits-loadPrevious ais-InfiniteHits-loadPrevious--disabled"`
+    );
+    expect(button.disabled).toEqual(true);
+    expect(showPrevious).not.toHaveBeenCalled();
```

Comment:
```suggestion
    expect(showPrevious).toHaveBeenCalledTimes(0);
```

This results in better error messages and is easier to refactor.